manuals/1.0/ja/study/03-behavioral/strategy-pattern.md

```
@@ -0,0 +1,163 @@
+            }
+        }
+        throw new InvalidArgumentException('Unknown shipping method');
+    }
+}
+```
+
+## パターンの本質
+
+Strategyパターンは明確な分離を生み出します：1つのクラスがすべてのアルゴリズムを知る代わりに、各アルゴリズムは自分自身だけを知ります。サービスはストラテジーのコレクションを受け取り、マッチが見つかるまで各ストラテジーの`supports()`メソッドを照会します。
+
+```
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)